addons/binding/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/serial/ValloxSerialInterface.java

```diff
@@ -0,0 +1,453 @@
+        telegram[1] = senderID;
+        telegram[2] = destination;
+        telegram[3] = variable;
+        telegram[4] = value;
+        telegram[5] = calculateChecksum(telegram);
+
+        logger.debug("sending telegram: {} {} {} {} {} {}", Telegram.byteToHex(telegram[0]),
+                Telegram.byteToHex(telegram[1]), Telegram.byteToHex(telegram[2]), Telegram.byteToHex(telegram[3]),
+                Telegram.byteToHex(telegram[4]), Telegram.byteToHex(telegram[5]));
+        serialWrite(telegram);
+    }
+
+    /**
+     * Send one telegram to the master unit.
+     * Example: send(Variable.FAN_SPEED.key, Telegram.convertBackFanSpeed((byte)4));
+     *
+     * @param variable one of the Variable keys in {@link Variable}}
+     * @param value the right binary coded value (see {@link Telegram} for some conversions)
+     * @throws IOException
+     */
+    public void send(byte variable, byte value) throws IOException {
+        send(variable, value, ValloxProtocol.ADDRESS_MASTER);
+    }
+
+    private void serialWrite(byte[] telegram) throws IOException {
+        try {
+            if (outputStream != null) {
+                outputStream.write(telegram);
+                outputStream.flush();
+            }
+        } catch (IOException e) {
+            String msg = "Exception writing to vallox: " + e.toString();
+            logger.error(msg);
+            for (StatusChangeListener l : this.getStatusListener()) {
+                l.statusChanged(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, msg);
+            }
+            throw e;
+        }
+    }
+
+    static byte calculateChecksum(byte[] pTelegram) {
+        int checksum = 0;
+        for (byte i = 0; i < pTelegram.length - 1; i++) {
+            checksum += pTelegram[i];
+        }
+        return (byte) (checksum % 256);
+    }
+
+    /**
+     * Start listening to telegrams on the serial interface. Stop if requested.
+     */
+    public void startListening() {
+        shutdownListening = false;
+        listenerExecutor = Executors.newSingleThreadExecutor();
+        listenerExecutor.submit(() -> {
+            while (!shutdownListening) {
+                try {
+                    Telegram telegram = receive();
+                    if (telegram == null) {
+                        // telegram too short or invalid
+                        continue;
+                    }
+                    logger.trace(telegram.toString());
+                    telegram.updateStore(vallox, vallueListener);
+                } catch (Exception e) {
+                    logger.error("Exception reading input stream: " + e.toString());
+                    try {
+                        Thread.sleep(2000);
+                    } catch (InterruptedException e1) {
+                        // ignore, simply retry, i.e. wait until reconnected
+                    }
+                }
+            }
+        });
+        logger.info("Start listening to vallox telegrams!");
```

Comment:
Please change the logging level to `debug`. `trace` can be used for even more details, where necessary.